packages/ui/src/components/date-input/date-input.tsx

```diff
@@ -3,7 +3,12 @@ import { PropsWithChildren, useEffect, useRef, useState } from "react";
 import Calendar from "../calendar";
 import { formatToMiddleEndian } from "./utils/date";
 
-export default function DateInput({ children }: PropsWithChildren) {
+interface DateInputProps {
+  id: string;
+  children: PropsWithChildren;
+}
+
+export default function DateInput({ id, children }: PropsWithChildren<DateInputProps>) {
```

Comment:
nit. I prefer we keep the parameter list for components like so: `{ children, ...props }`, because, otherwise, it can get hard to read when there are many props. We can then destructure props like `const { id } = props;`.